code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/github/GithubTokenCreator.java

```diff
@@ -25,34 +30,35 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
-
 import javax.inject.Inject;
 
 import org.apache.commons.lang3.ObjectUtils;
 
-import com.netflix.config.DynamicBooleanProperty;
 import com.netflix.config.DynamicLongProperty;
 
-public class GithubTokenHandler implements TokenHandler {
-
-    @Inject
-    private TokenService tokenService;
-    @Inject
-    private AuthDao authDao;
-    @Inject
-    private GithubClient githubClient;
+public class GithubTokenCreator extends GithubConfigurable implements TokenCreator {
 
     private static final DynamicLongProperty TOKEN_EXPIRY_MILLIS = ArchaiusUtil.getLong("api.auth.jwt.token.expiry");
-    private static final DynamicBooleanProperty SECURITY = ArchaiusUtil.getBoolean("api.security.enabled");
-
     @Inject
     ProjectResourceManager projectResourceManager;
     @Inject
     ObjectManager objectManager;
     @Inject
     GithubUtils githubUtils;
+    @Inject
+    GithubClient githubClient;
+    @Inject
+    private TokenService tokenService;
+    @Inject
+    private AuthDao authDao;
+    @Inject
+    private GithubIdentitySearchProvider githubIdentitySearchProvider;
```

Comment:
@hibooboo2 As a general java programming comment, its best to have these injected member variables be interfaces and not concrete classes. This improves encapsulation, enforces strong contracts, prevents leaky abstractions, and enables other programming paradigms that rely on interfaces to work correctly (like AOP and method interception for transaction management or dynamic mocking libraries).

In this particular case, you call githubIdentitySearchProvider. getAccessToken(), which is not part of the IdentitySearchProvider interface and probably doesn't make sense to be on an identity search provider in general.

You don't have to change for this PR unless you really want to because it doesn't break anything how it is, but keep it in mind in the future.